lib/compress/zstd_compress.c

```diff
@@ -1203,42 +1204,74 @@ static size_t ZSTD_resetCCtx_usingCDict(ZSTD_CCtx* cctx,
                             U64 pledgedSrcSize,
                             ZSTD_buffered_policy_e zbuff)
 {
+    /* We have a choice between copying the dictionary context into the working
+     * context, or referencing the dictionary context from the working context
+     * in-place. We decide here which strategy to use. */
+    /* TODO: pick reasonable cut-off size, handle ZSTD_CONTENTSIZE_UNKNOWN */
+    int attachDict = ( pledgedSrcSize <= 8 KB
```

Comment:
So, 8 KB is the final selected cut-off threshold ?
Do you have graphs / data showing what happens when no cut-off is done, or done later ?

Reply:
For the time being, yes. As I work with the other levels, we'll see if it needs to be level-dependent or whether 8 KB is best for all levels. But that seemed to be the crossover point in perf at the time when I was setting that up. I got rid of those graphs, but I'll kick off a new pass now with no cut-off so we can see.